src/main/scala/zio/ftp/settings.scala

```diff
@@ -0,0 +1,107 @@
+package zio.ftp
+
+import java.net.Proxy
+
+import net.schmizz.sshj.{ Config => SshConfig, DefaultConfig => DefaultSshConfig }
+import org.apache.commons.net.ftp.FTPClient
+
+sealed trait credentials {
+
+  sealed trait FtpCredentials {
+    val username: String
+    val password: String
+  }
+
+  def credentials(_username: String, _password: String): FtpCredentials = new FtpCredentials {
+    val username: String = _username
+    val password: String = _password
+  }
+}
+
+object settings extends credentials {
+
+  final case class SFtpSettings(
+    host: String,
+    port: Int,
+    credentials: FtpCredentials,
+    strictHostKeyChecking: Boolean,
+    knownHosts: Option[String],
+    sftpIdentity: Option[SftpIdentity],
+    sshConfig: SshConfig
+  )
+
+  object SFtpSettings {
+
+    def apply(host: String, port: Int, creds: FtpCredentials): SFtpSettings =
+      new SFtpSettings(
```

Comment:
Can omit `new` here because you get `apply` method for free in companion object of `SFtpSettings`.